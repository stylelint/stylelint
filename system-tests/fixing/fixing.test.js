/* @flow */
"use strict"

const systemTestUtils = require("../systemTestUtils")
const stylelint = require("../../lib")

it("fixing", () => {
  return stylelint.lint({
    files: [systemTestUtils.caseStylesheetGlob("fixing")],
    configFile: systemTestUtils.caseConfig("fixing"),
    fix: true,
  }).then((output) => {
    expect(systemTestUtils.prepResults(output.results)).toMatchSnapshot()
    expect(output.results[0]._postcssResult.root.toResult().css).toMatchSnapshot()
  })
})
