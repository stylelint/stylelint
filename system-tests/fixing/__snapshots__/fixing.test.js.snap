// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixing 1`] = `
Array [
  Object {
    "deprecations": Array [],
    "errored": true,
    "ignored": undefined,
    "invalidOptionWarnings": Array [],
    "source": "fixing/stylesheet.css",
    "warnings": Array [
      Object {
        "column": 1,
        "line": 7,
        "rule": "comment-no-empty",
        "severity": "error",
        "text": "Unexpected empty comment (comment-no-empty)",
      },
    ],
  },
]
`;

exports[`fixing 2`] = `
"a { @extend placeholder; }
a {} @media {}
@charset 'UTF-8';
a { color: #000; }
a {}

/* comment */

/**/

/*comment*/

/* TODO: */
@custom-media --bar (min-width: 30em);
a {
  & .foo { /* 1 */
    &__foo { /* 2 */
      & > .bar {} /* 3 */
    }
  }
}
"
`;
